### ISSUE TITLE
v1.5.1 crash on KrylovKit on Windows

### ISSUE THREAD

**Issue by Roger-luo** [JuliaLang/julia#37306]
The following code works fine on 1.4 but crashes on 1.5 and nightly on Windows. It works fine on Mac OS/Linux on 1.5 however.

```jl
using Yao
using YaoExtensions
using KrylovKit
using SparseArrays
# 
hami = heisenberg(14)
m = mat(hami)
sp2 = SparseMatrixCSC(SparseMatrixCSC(m')')

function failing_function(sp)
    size(sp)
    length(sp)
    eigsolve(sp, 1, :SR)
end

failing_function(sp2)
```

The error msg looks like the following

```jl
Please submit a bug report with steps to reproduce this fault, and any error messages that follow (in their entirety). Thanks.
Exception:
Please submit a bug report with steps to reproduce this fault, and any error messages that follow (in their entirety). Thanks.
Exception: EXCEPTION_ACCESS_VIOLATION at 0x19b3d5c0 --  at 0x19b3d5c0 -- OLATION with steps to reproduce this fault, and any error messages that follow (in their entirety). Thanks.
Exception: EXCEPTION_ACCESS_VIOLATION at 0x19b3d5c0 -- OLATION with steps to reproduce this fault, and any error messages that follow (in their entirety). Thanks.
Exception: EXCEPTION_ACCESS_VIOLATION at 0x19b3d5c0 --  at 0x19b3d5c0 -- OLATION with steps to reproduce this fault, and any error messages that follow (in their entirety). Thanks.
Exception: EXCEPTION_ACCESS_VIOLATION at 0x19b3d5c0 --  at 0x19b3d5c0 -- OLATION with steps to reproduce this fault, and any error messages that follow (in their entirety). Thanks.
Exception: EXCEPTION_ACCESS_VIOLATION at 0x19b3d5c0 --  at 0x19b3d5c0 -- OLATION at 0x19b3d5c0 --  at 0x19b3d5c0 --  at 0x19b3d5c0 -- OLATION with steps to reproduce this fault, and any error messages that follow (in their entirety). Thanks.
Exception: EXCEPTION_ACCESS_VIOLATION at 0x19b3d5c0 --  at 0x19b3d5c0 -- OLATION with steps to reproduce this fault, and any error messages that follow (in their entirety). Thanks.
Exception: EXCEPTION_ACCESS_VIOLATION at 0x19b3d5c0 --
Please submit a bug report with steps to reproduce this fault, and any error messages that follow (in their entirety). Thanks.
Exception:
```

## Version Info

```jl
julia> versioninfo()
Julia Version 1.5.1
Commit 697e782ab8 (2020-08-25 20:08 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: AMD Ryzen 9 3900X 12-Core Processor
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-9.0.1 (ORCJIT, znver2)
```

```jl
julia> versioninfo()
Julia Version 1.6.0-DEV.776
Commit 5f0469b227 (2020-08-31 10:52 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: AMD Ryzen 9 3900X 12-Core Processor
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-9.0.1 (ORCJIT, znver2)
```

XRef: https://github.com/QuantumBFS/Yao.jl/issues/256

**Comment by maartenvd** [comment]
possibly #36976 ?

**Comment by Roger-luo** [comment]
this seems to be fixed on 1.7 now
